### PR TITLE
Vasyl chorney

### PR DESCRIPTION
**Pull request by VChorney**

### CODE REVIEW

**Comment by VChorney** [comment]
I'd like to add validation and little more refactoring

File: src/com/home_work/spring_boot/ui/UIRepresentation.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+@Component
+public class UIRepresentation {
+
+    private ValidationService validationService;
+    private MailService mailService;
+    private Letter letter = new Letter();
+    ExecutorService executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
```
Comment (inventorsoft-academy) [inline src/com/home_work/spring_boot/ui/UIRepresentation.java]: this should be created in separate configuration. do not initialize your dependencies, let spring do it

File: src/com/home_work/spring_boot/ui/UIRepresentation.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        try {
+            while (answer.equals("yes")) {
+                initLetter(letter, scanner);
+                validate(letter);
+                mailService.saveMail(letter);
+                letters = mailService.getMailsToSend();
+                answer = askForSendLetter(scanner);
+                executorService.execute(new MailSenderThread(letters.get(letters.size() - 1), mailService));
+            }
+            executorService.shutdown();
```
Comment (inventorsoft-academy) [inline src/com/home_work/spring_boot/ui/UIRepresentation.java]: UI component shouldn't handle lifecycle of another component.

File: src/com/home_work/spring_boot/ui/UIRepresentation.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        System.out.println("Enter delivery time(yyyy-MM-dd HH:mm)");
+        letter.setDeliveryTime(parseToLocalDateTime(scanner.nextLine()));
+    }
+
+    private LocalDateTime parseToLocalDateTime(String deliveryTime) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        return LocalDateTime.parse(deliveryTime, formatter);
+    }
+
+    private void checkMailsForSend() {
```
Comment (inventorsoft-academy) [inline src/com/home_work/spring_boot/ui/UIRepresentation.java]: Move sending of the emails logic into separate component, it's not responsibility of ui

**Review by inventorsoft-academy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**